assets/src/latest-stories-block/block/edit.js

```diff
@@ -144,7 +145,27 @@ const LatestStoriesEdit = ({ attributes, setAttributes }) => {
     debouncedFetchLatestStories();
   }, [numOfStories]); /* eslint-disable-line react-hooks/exhaustive-deps */
 
-  const willShowStoryPoster = 'grid' != viewType ? true : isShowingStoryPoster;
+  useEffect(() => {
+    if ('circles' !== viewType) {
+      setAttributes({
+        isShowingStoryPlayer: false,
+        isShowingTitle: true,
+        isShowingAuthor: true,
+        isShowingDate: true,
+      });
+    }
+
+    if ('circles' === viewType) {
+      setAttributes({
+        isShowingStoryPlayer: false,
+        isShowingTitle: true,
+      });
+    }
+  }, [viewType, setAttributes]);
+
+  const willShowStoryPlayer =
+    'grid' !== viewType ? false : isShowingStoryPlayer;
+
   const willShowDate = 'circles' === viewType ? false : isShowingDate;
   const willShowAuthor = 'circles' === viewType ? false : isShowingAuthor;
```

Comment:
With this logic in place, I don't think we need these checks while rendering as they will always be based on the view type, did I understand it right? In short, can these checks `willShowDate`, `willShowAuthor`, etc can not be removed now? Also, should this useEffect not be moved to `latestStoriesControls.js` file as that's where these flags are being handled?

Reply:
True, these checks are based on the view then we don't need them as we always set/update the state on view change.

Also, I feel keeping the state control in the main file is much easier to understand the block state logic.